core/src/main/java/org/apache/accumulo/core/file/blockfile/cache/BlockCacheConfiguration.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.file.blockfile.cache;
+
+import java.util.Map;
+
+import org.apache.accumulo.core.conf.AccumuloConfiguration;
+import org.apache.accumulo.core.conf.Property;
+
+public class BlockCacheConfiguration {
+
+  public static final String MAX_SIZE_PROPERTY = "max.size";
+  public static final String BLOCK_SIZE_PROPERTY = "block.size";
+
+  private static final Long DEFAULT = Long.valueOf(-1);
+
+  /** Maximum allowable size of cache (block put if size > max, evict) */
+  private final long maxSize;
+
+  /** Approximate block size */
+  private final long blockSize;
+
+  protected final BlockCacheConfigurationHelper helper;
+
+  public BlockCacheConfiguration(AccumuloConfiguration conf, CacheType type, BlockCacheFactory<?,?> factory) {
+
+    helper = new BlockCacheConfigurationHelper(conf, type, factory);
+
+    Map<String,String> props = conf.getAllPropertiesWithPrefix(Property.GENERAL_ARBITRARY_PROP_PREFIX);
+    this.maxSize = getOrDefault(props, helper.getFullPropertyName(MAX_SIZE_PROPERTY), DEFAULT);
```

Comment:
I suspect this property is specific to the LruCache.  I looked at the TinyLFu impl and it uses it in a round about way.  May want to consider deprecating and replacing with a specific lru config.   For TinyLfu, maybe it would be better to have config for initial capacity size rather than use this block size.

I am wondering for these cache specif configs where the best place to document them is.